### ISSUE TITLE
Refactor exection of runs

### ISSUE THREAD

**Issue by vpetersson** [WoTTsecurity/agent#249]
**tl;dr**:  We need more granular control over what's run and when. Some checks need to run often, others not very frequently.

I want to have three different intervals for various checks:

 * Every five minutes
 * Every hour
 * Every day

Here's a draft for the frequency of the runs:

| Function        | Frequency           | Trigger  |
| ------------- |-------------| -----|
| device_operating_system_version: platform.release() | Daily |
| fqdn: socket.getfqdn() | Daily |
| ipv4_address: get_primary_ip() | Hourly |
| uptime: get_uptime() | Hourly |
| agent_version: str(__version__) | Hourly |
| installation: detect_installation().name | Daily |
| os_release: rpi_helper.get_os_release() | Daily |
| packages: get_deb_packages() | Hourly |
| connections ports: security_helper.netstat_scan() | Hourly |
| blocklist: ping | Daily | 
| iptables_helper.block(blocklist) | Daily | Boot |
| processes: security_helper.process_scan() | Hourly |
| logins: journal_helper.logins_last_hour() | Hourly |
| firewall_rules: iptables_helper.dump() | Hourly | Change in Dash |
| scan_info: ports | Hourly |
| netstat: connections | Hourly |
| selinux_status: security_helper.selinux_status() | Daily |
| app_armor_enabled: security_helper.is_app_armor_enabled() | Daily |
| default_password: security_helper.check_for_default_passwords(CONFIG_PATH) | Daily |
| audit_files: security_helper.audit_config_files() | Hourly | 
| auto_upgrades: rpi_helper.auto_upgrades_enabled() | Hourly |
| mysql_root_access: security_helper.mysql_root_access() | Hourly |
| kernel_package: rpi_helper.kernel_deb_package() | Hourly |
| cpu: security_helper.cpu_vulnerabilities() | Hourly |
| device_manufacturer: Raspberry Pi | Daily |
| device_model: rpi_metadata[hardware_model] | Daily |

In the future, we will also need support for events/functions triggered from the server.
